haskell-tests/Types/Generators.hs

```diff
@@ -105,130 +106,146 @@ genUrlText =
-        icParam <- genParameter
-        return InitContract{..}
-
-    genUpdate = do
-        uAmount <- Amount <$> arbitrary
-        uAddress <- genCAddress
-        uMessage <- genParameter
-        uReceiveName <- genReceiveName
-        return Update{..}
-
-    genTransfer = do
-        a <- genAddress
-        amnt <- Amount <$> arbitrary
-        return $ Transfer a amnt
-
-    genAddBaker = do
-        abElectionVerifyKey <- VRF.publicKey <$> arbitrary
-        abSignatureVerifyKey <- BlockSig.verifyKey <$> genBlockKeyPair
-        (abAggregationVerifyKey, abProofAggregation) <- genAggregationVerifyKeyAndProof
-        abProofSig <- genDlogProof
-        abProofElection <- genDlogProof
-        abBakingStake <- arbitrary
-        abRestakeEarnings <- arbitrary
-        return AddBaker{..}
-
-    genRemoveBaker = return RemoveBaker
-
-    genUpdateBakerStake =
-        UpdateBakerStake <$> arbitrary
-
-    genUpdateBakerRestakeEarnings =
-        UpdateBakerRestakeEarnings <$> arbitrary
-
-    genUpdateBakerKeys = do
-        ubkElectionVerifyKey <- VRF.publicKey <$> arbitrary
-        ubkSignatureVerifyKey <- BlockSig.verifyKey <$> genBlockKeyPair
-        (ubkAggregationVerifyKey, ubkProofAggregation) <- genAggregationVerifyKeyAndProof
-        ubkProofSig <- genDlogProof
-        ubkProofElection <- genDlogProof
-        return UpdateBakerKeys{..}
-
-    genUpdateCredentialKeys = do
-        uckKeys <- genCredentialPublicKeys
-        uckCredId <- genCredentialId
-        return UpdateCredentialKeys{..}
-
-    genTransferToEncrypted =
-        TransferToEncrypted . Amount <$> arbitrary
-
-    genRegisterData = do
-        n <- chooseInt (0, maxRegisteredDataSize)
-        rdData <- RegisteredData . BSS.pack <$> vectorOf n arbitrary
-        return RegisterData{..}
-
-    genConfigureBaker = do
-        cbCapital <- arbitrary
-        cbRestakeEarnings <- arbitrary
-        cbOpenForDelegation <- liftArbitrary $ elements [OpenForAll, ClosedForNew, ClosedForAll]
-        cbSignatureVerifyKey <-
-            liftArbitrary $
-                (,) <$> (BlockSig.verifyKey <$> genBlockKeyPair) <*> genDlogProof
-        cbElectionVerifyKey <-
-            liftArbitrary $
-                (,) <$> (VRF.publicKey <$> arbitrary) <*> genDlogProof
-        cbAggregationVerifyKey <- liftArbitrary genAggregationVerifyKeyAndProof
-        cbMetadataURL <- liftArbitrary genUrlText
-        cbTransactionFeeCommission <- liftArbitrary genRewardFraction
-        cbBakingRewardCommission <- liftArbitrary genRewardFraction
-        cbFinalizationRewardCommission <- liftArbitrary genRewardFraction
-        return ConfigureBaker{..}
-
-    genConfigureDelegation = do
-        cdCapital <- arbitrary
-        cdRestakeEarnings <- arbitrary
-        cdDelegationTarget <-
-            liftArbitrary $
-                oneof [return DelegateToLPool, DelegateToBaker . BakerId . AccountIndex <$> arbitrary]
-        return ConfigureDelegation{..}
+
```

Comment:
I have taken these generators out of the where clause, since I needed some of them for the negative tests in `PayloadSerializationSpec.hs`. Any better idea?